crates/re_viewer/src/app.rs

```diff
@@ -374,16 +374,22 @@ impl App {
                 store_hub.clear_blueprint();
             }
             SystemCommand::UpdateBlueprint(blueprint_id, updates) => {
-                let blueprint_db = store_hub.entity_db_mut(&blueprint_id);
-                for row in updates {
-                    match blueprint_db.add_data_row(row) {
-                        Ok(()) => {}
-                        Err(err) => {
-                            re_log::warn_once!("Failed to store blueprint delta: {err}");
+                if !self.state.app_options.show_blueprint_timeline {
```

Comment:
can you explain this `if` (with a comment)?

Why are we ignoring blueprint updates if the blueprint timeline is visible in the UI?

Reply:
> Why are we ignoring blueprint updates if the blueprint timeline is visible in the UI?

Good call -- both a comment and a TODO.

In short the answer to this is because when the blueprint panel is visible, the blueprint time-selection now changes which version of blueprint state is used to drive the UI. If you then try to modify the blueprint in some way, the edits you're making are relative to a historical blueprint state, but applied to the end of the blueprint timeline, potentially putting things in an unexpected state.

Given this (and partially answering the question below) I think we might want to call this whole thing "Blueprint inspection mode" with an explanation of the fact that the blueprint can't be edited while being inspected.

The one caveat to this (hence the TODO to be written) would be that this logic could be relaxed if we are in follow mode. However, I wasn't positive that follow mode gives us the guarantee that the query time is always the latest data without the possibillity of a 1-frame delay.